### ISSUE TITLE
Word2vec 里面的 random_softmax_loss 好像有问题

### ISSUE THREAD

**Issue by imhuay** [bojone/tf_word2vec#1]
```
    def random_softmax_loss(self, nb_negative, inputs, targets, weights, biases=None):  # 定义loss函数：随机抽样、点乘、交叉熵。
        nb_classes, real_batch_size = tf.shape(weights)[0], tf.shape(targets)[0]
        negative_sample = tf.random_uniform([real_batch_size, nb_negative], 0, nb_classes, dtype=tf.int32)
        random_sample = tf.concat([targets, negative_sample], axis=1)
        sampled_weights = tf.nn.embedding_lookup(weights, random_sample)
        if biases:
            sampled_biases = tf.nn.embedding_lookup(biases, random_sample)
            sampled_logits = tf.matmul(inputs, sampled_weights, transpose_b=True)[:, 0, :] + sampled_biases
        else:
            sampled_logits = tf.matmul(inputs, sampled_weights, transpose_b=True)[:, 0, :]
        sampled_labels = tf.zeros([real_batch_size], dtype=tf.int32)
        return tf.reduce_mean(
            tf.nn.sparse_softmax_cross_entropy_with_logits(logits=sampled_logits, labels=sampled_labels))

    def build_model(self):  # 建立模型，比较常规
        self.sess = tf.Session()
        self.embeddings = tf.Variable(tf.random_uniform([self.total_words + 1, self.word_size], -0.05, 0.05))
        self.normalized_embeddings = tf.nn.l2_normalize(self.embeddings, 1)
        self.target_words = tf.placeholder(tf.int32, shape=[None, 1])
        if self.model == 'cbow':
            self.input_words = tf.placeholder(tf.int32, shape=[None, 2 * self.window])
            self.input_vecs = tf.nn.embedding_lookup(self.embeddings, self.input_words)
            self.input_vecs = tf.expand_dims(tf.reduce_sum(self.input_vecs, 1), 1)
        else:
            self.input_words = tf.placeholder(tf.int32, shape=[None, 1])
            self.input_vecs = tf.nn.embedding_lookup(self.embeddings, self.input_words)
        if self.shared_softmax:
            self.loss = self.random_softmax_loss(self.nb_negative, self.input_vecs, self.target_words, self.embeddings)
        else:
            self.softmax_weights = tf.Variable(tf.random_uniform([self.total_words + 1, self.word_size], -0.05, 0.05))
            self.softmax_biases = tf.Variable(tf.zeros([self.total_words + 1]))
            self.loss = self.random_softmax_loss(self.nb_negative, self.input_vecs, self.target_words,
                                                 self.softmax_weights, self.softmax_biases)
```

`sampled_labels = tf.zeros([real_batch_size], dtype=tf.int32) ` 这里为什么没有加入正例？
按我的理解，这里第一维应该是 1，因为前面 `random_sample = tf.concat([targets, negative_sample], axis=1)` 把target 也拼进去了

**Comment by GaoQ1** [comment]
估计你已经解决了，刚看的时候这里我刚才也愣了一下，看了tf.nn.sparse_softmax_cross_entropy_with_logits这个方法，是labels是稀疏表示，这儿tf.zeros意思就是分类为第一类，(0,0,0..)也就是正例。

**Comment by ghost** [comment]
基于python啥版本呀 我报错语法错误 ：python dos Missing parenthess in call to 'print'
